core-bundle/src/EventListener/DataContainer/ToggleNodesLabelListener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\DataContainer;
+use Contao\DC_Table;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\Attribute\AttributeBagInterface;
+
+class ToggleNodesLabelListener
+{
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(RequestStack $requestStack)
+    {
+        $this->requestStack = $requestStack;
+    }
+
+    public function __invoke(string $table): void
+    {
+        if (
+            DC_Table::class !== DataContainer::getDriverForTable($table)
+            || !isset($GLOBALS['TL_DCA'][$table]['list']['global_operations']['toggleNodes'])
+            || isset($GLOBALS['TL_DCA'][$table]['list']['global_operations']['toggleNodes']['label'])
+            || 'ptg=all' !== ($GLOBALS['TL_DCA'][$table]['list']['global_operations']['toggleNodes']['href'] ?? null)
+            || null === ($request = $this->requestStack->getCurrentRequest())
+            || null === ($session = $request->getSession())
```

Comment:
The session should be retrieved from the request stack.

Reply:
That is not available in Symfony 4.4 (which we use for Contao 4.9)

Reply:
Then please create a second PR for Contao 5 as the current changes are not upstream compatible. (Or only add this feature in Contao 5.1, which I would prefer.)

Reply:
I can do that as soon as this is merged so we know the final implementation, right?